SDWebImage/Core/SDImageTransformer.h

```diff
@@ -38,10 +47,10 @@ FOUNDATION_EXPORT NSString * _Nullable SDTransformedKeyForKey(NSString * _Nullab
  Transform the image to another image.
 
  @param image The image to be transformed
- @param key The cache key associated to the image
+ @param key The cache key associated to the image. This arg is a hint for image source, not always useful and should be nullable. In the future we will remove this arg.
  @return The transformed image, or nil if transform failed
  */
-- (nullable UIImage *)transformedImageWithImage:(nonnull UIImage *)image forKey:(nonnull NSString *)key;
+- (nullable UIImage *)transformedImageWithImage:(nonnull UIImage *)image forKey:(nonnull NSString *)key API_DEPRECATED("The key arg will be removed in the future. Update your code and don't rely on that.", macos(10.10, API_TO_BE_DEPRECATED), ios(8.0, API_TO_BE_DEPRECATED), tvos(9.0, API_TO_BE_DEPRECATED), watchos(2.0, API_TO_BE_DEPRECATED));
```

Comment:
Can we change the method signature to `nullable` for this key arg ?

Reply:
It's okey to change in `nullable`.

Reply:
Seems change it to `nullable` will cause API break for exist custom protocol for Swift User. (Objective-C does not cause issue). So let keep it until 6.0.0